src/Service/S3/src/Input/PutObjectRequest.php

```diff
@@ -541,7 +536,7 @@ public function requestHeaders(): array
             $headers['Content-Language'] = $this->ContentLanguage;
         }
         if (null !== $this->ContentLength) {
-            $headers['Content-Length'] = $this->ContentLength;
+            $headers['Content-Length'] = (string) $this->ContentLength;
```

Comment:
I make sure headers always are strings and not integers. This is according to RFC 7230

Reply:
shouldn't it be done in `Request` object?

Reply:
Oops. I merged too fast.. 

Yeah.. maybe.  But the $header annotation on the request object says there should be strings already.